### PR TITLE
[ALOY-1346] Controller template should no longer declare args variable

### PR DESCRIPTION
**Pull request by FokkeZB**
https://jira.appcelerator.org/browse/ALOY-1346

### CODE REVIEW

**Comment by skypanther** [comment]
This feature was added (by me) in response to the request of two of your solutions engineers. I would make sure you're not removing something that your customers have implemented and expect.

**Comment by FokkeZB** [comment]
Thanks for the heads up @skypanther! I assume that was before we exposed it via `$.args`? I'll ask SE on their opinion about removing this.

**Comment by rblalock** [comment]
I'm for replacing it with `var args = $.args;`.  Leaving that stubbed out is helpful for beginners.

**Comment by FokkeZB** [comment]
Agree with @rblalock @feons, mostly because it won't break (copied) code and expectations. Updated the PR.

**Comment by grantges** [comment]
I'm for actually removing the reference - and replacing it with a comment at the top of the controller file.

// Arguments passed into this controller can be accessed off of the '$.args' object. 
//
// Example
//     myProperty = $.args.myProperty

**Comment by rblalock** [comment]
@grantges ewww

**Comment by FokkeZB** [comment]
```
// Arguments passed into this controller can be accessed off of the '$.args' object. 
//
// Example
// myProperty = $.args.myProperty
//
// You can add a class to a proxy via `$.addClass`.
//
// Example
// $.addClass($.someView, "someclass");
//
// ...
```

😜

**Comment by rblalock** [comment]
I wonder how many people  use the stubbed out controllers?  The last time I did was when...it was in beta :-D

**Comment by rblalock** [comment]
It'd be good to have a stubbed out controller in our docs that has all these comments and such.  e.g. https://facebook.github.io/react/docs/reusable-components.html

**Comment by FokkeZB** [comment]
Yes, I'd prefer that instead of commenting just this one feature in the stub.

**Comment by FokkeZB** [comment]
@feons let's go with the change as it is now.

**Comment by skypanther** [comment]
A new controller (`alloy generate controller <name>`) already gives you the following at the top of the file

```
var args = arguments[0] || {};
```

All the original change did was expose those args such they could be used in TSS style if statements. It added `args` as a property of `$`. There's no real value to a dev to have them available as `$.args` vs that var statement other than maybe some convenience for inexperienced devs that don't know what the var statement is doing.

**Comment by FokkeZB** [comment]
@skypanther well, I think having `$.args` as a documented way to get args instead of having to do `var args = arguments[0] || {}` in every controller has value, don't you agree?

**Comment by skypanther** [comment]
yeah, that would be fine by me. Honestly, I had forgotten that was in there and add the `var args...` statement myself every time I create a controller by hand.

**Comment by FokkeZB** [comment]
Same for me :wink:

**Comment by FokkeZB** [comment]
Updated with comment above `var args = $.args`

**Comment by feons** [comment]
*LGTM*

File: Alloy/template/controller.js [hunk 1]
```diff
@@ -1 +1,2 @@
-var args = arguments[0] || {};
\ No newline at end of file
+// Arguments passed into this controller can be accessed off of the `$.args` object directly or:
```
Comment (FokkeZB) [inline Alloy/template/controller.js]: Mmm... "off of"?

Comment (ewieberappc) [inline Alloy/template/controller.js]: I also dislike the "off of" wording. Perhaps "by" would read better?

Comment (FokkeZB) [inline Alloy/template/controller.js]: Or "accessed via" ?

Comment (ewieberappc) [inline Alloy/template/controller.js]: works for me

**Comment by FokkeZB** [comment]
Changed :D

**Comment by FokkeZB** [comment]
? that didn't go well. Looks like I have two patch-12 branches now. Will resolve tomorrow

**Comment by FokkeZB** [comment]
@ewieberappc somehow I done something wrong with the branch. Here's an additional PR to be merged after this with the wording:

https://github.com/appcelerator/alloy/pull/753

@feons go ahead ;)

**Pull request merged**